### DISCUSSION TITLE
Browser-based interface

### DISCUSSION THREAD

**Discussion by ai-sroth** [atom/atom#22398] [Q&A]
Is there a way to set up a browser-based interface to access an installation of atom on a remote server? I currently use RStudio server as my IDE for work with an R installation running on an AWS ubuntu server and am wondering if it would be possible to do the same with Atom to take advantage of its customizability while allowing my team to access it remotely on the same server. My goal is to then set up teletype so that coders working remotely in different locations can collaborate on projects using the AWS server installation. I love working in Atom on my local for personal projects, so thanks for maintaining such a great open-source editor!

**Comment by ThatXliner** [comment]
See https://github.com/atom/atom/discussions/22137 and https://flight-manual.atom.io/faq/sections/atom-in-the-cloud/

**Reply by ai-sroth** [reply]
Ah, got it. Looks like I was using some poor search terms before. Thanks for pointing me to these resources.

**Comment by datavectors** [comment]
It is quite feasible as [this service](https://www.rollapp.com/search?query=Atom) illustrates.
You can test one app at a time with Free licence but you have to upgrade to Premium licence to mix Atom with other apps in a persistent workflow. I tried with a monthly licence.
I was not comfortable putting all into an unknown site (although it is fine for educational use) and if I were to follow this route I would emulate a remote desktop on a private image of Ubuntu in the cloud.
However RStudio is not supported in Rollapp if you search the list of enabled apps.

**Reply by ai-sroth** [reply]
Thanks for the suggestion! This could work for my case.

**Comment by datavectors** [comment]
When I tested this approach (which works better with a Premium monthly plan) I used a free Dropbox account to share common files.

Another approach worth mentioning (in concert with rollApp) is to use desktop automation scripts to bind desktop operations to cloud services. For this I use Actiona (version 3.10.1) and Albert. 

The common language across these services (local and browser) is simply the keyboard and mouse clicks and object navigation (menus) can be emulated/automated in automation scripts.  See what xdotool offers in Ubuntu. I also can access Actiona from Atom through installed package process-palette and I have written in the old Atom forum about using toolchains in this way. It is similar to the approach to use "smart connectors" between "microservices".   Atom and RStudio can thus be linked in workflows.

**Reply by ThatXliner** [reply]
Wait. Maybe I didn't understand but did you mean by automating actions?

E.g. a click in the browser is a click on the server's computer

**Reply by datavectors** [reply]
> but did you mean by automating actions?

Yes.  But you will have to wait a while for me to post a fuller explanation or workflow .. and possibly a demonstrator in Heroku.
Meanwhile ..

> E.g. a click in the browser is a click on the server's computer

Yes. But who/what executes the click?  A human or a bot?
You are assuming that the desktop user performs the clickthrough.

What if (in reverse) the server wants permission to address (autodrive) the user's desktop applications (such as Atom?).  Or to request permission to conduct a workflow?

One spinoff is that we can drop the ideas of giving screenshots in YouTube tutorials.  Instead the actual code is presented in the user's desktop (subject always to permissions granted and security policy being covered).

**Reply by ThatXliner** [reply]
I was thinking, for a cloud idea, is to simulate Atom (as a webpage) on the server's file system. Kind of like gitpod.io or something

**Comment by hapylestat** [comment]
> Is there a way to set up a browser-based interface to access an installation of atom on a remote server? I currently use RStudio server as my IDE for work with an R installation running on an AWS ubuntu server and am wondering if it would be possible to do the same with Atom to take advantage of its customizability while allowing my team to access it remotely on the same server. My goal is to then set up teletype so that coders working remotely in different locations can collaborate on projects using the AWS server installation. I love working in Atom on my local for personal projects, so thanks for maintaining such a great open-source editor!

you could drop ideas about poor Atom, and better look to VSCode-server or Theia, which have  full browser support.

**Reply by ThatXliner** [reply]
While I *do* use VSCode and Gitpod.io sometimes, some people may find Atom more simple and minimal. ~~And more beautiful, too~~

**Reply by ai-sroth** [reply]
lol, I do really like atom and have found a mostly satisfactory solution using the ftp-remote-edit package, but I'll consider a VSCode-server-based approach as well in case it offers additional benefits.
